src/Controller/SitemapController.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace OHMedia\PageBundle\Controller;
+
+use OHMedia\PageBundle\Repository\PageRepository;
+use OHMedia\PageBundle\Sitemap\AbstractSitemapUrlProvider;
+use OHMedia\PageBundle\Sitemap\SitemapUrl;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+class SitemapController extends AbstractController
+{
+    private array $sitemapUrlProviders = [];
+
+    public function addSitemapUrlProvider(AbstractSitemapUrlProvider $sitemapUrlProvider): void
+    {
+        $this->sitemapUrlProviders[] = $sitemapUrlProvider;
+    }
+
+    #[Route('/sitemap.xml', name: 'oh_media_page_sitemap')]
+    public function sitemap(PageRepository $pageRepository): Response
+    {
+        $pages = $pageRepository->createQueryBuilder('p')
+            ->where('p.homepage = 1 OR p.sitemap = 1')
+            ->orderBy('p.order_global', 'asc')
+            ->getQuery()
+            ->getResult();
+
+        $sitemapUrls = [];
+
+        foreach ($pages as $page) {
+            if (!$page->isVisibleToPublic()) {
+                continue;
+            }
+
+            $pageRevision = $page->getCurrentPageRevision();
```

Comment:
Reusing logic through functions rather than trying to bake it into the query. This page doesn't need to be optimized.